gui/IrodsBrowser.py

```diff
@@ -142,10 +142,18 @@ def _fill_resources_tab(self, obj_path):
         if self.ic.dataobject_exists(obj_path):
             obj = self.ic.get_dataobject(obj_path)
             hierarchies = [repl.resc_hier for repl in obj.replicas]
+            replicas = [resc.resource_name for resc in obj.replicas]
             self.resourceTable.setRowCount(len(hierarchies))
             for row, hierarchy in enumerate(hierarchies):
                 self.resourceTable.setItem(
                     row, 0, PyQt6.QtWidgets.QTableWidgetItem(hierarchy))
+                for repl in replicas:
+                    if repl in hierarchy:
+                        item = PyQt6.QtWidgets.QTableWidgetItem()
+                        item.setCheckState(PyQt6.QtCore.Qt.CheckState.Checked)
+                        item.setFlags(PyQt6.QtCore.Qt.ItemFlag.ItemIsEnabled)
+                        self.resourceTable.setItem(row, 1, item)
```

Comment:
My understanding is that `resource_name` is _always_ in `resc_hier`.  It is the last element in the semi-colon delimited list of resource names.  When the hierarchy is constructed for any data object, this name is included as it is the only resource name actually stored in the table describing the data objects.

```python
if repl in hierarchy:
```

should be superfluous.  I'd be curious to see what errors occur when that line is removed.

Reply:
Yes you are right. When we extract the resources an object is stored on it is always present in the hierarchy. The if statement comes from a copy paste from setting the default resource. 
Can be safely deleted.